feat(treesitter): add query_linter from nvim-treesitter/playground (#22784)
neovim/neovim: runtime/doc/news.txt

The following new APIs or features were added.

• Added automatic linting of tree-sitter query files (see |ft-query-plugin|).

>   automatic linting can be turned off via >
>       :let g:query_lint_on = []
> <

Reviewer: ```suggestion
  automatic linting can be turned off via >lua
      vim.g.query_lint_on = nil
<
```

• |vim.iter()| provides a generic iterator interface for tables and Lua
iterators |luaref-in|.

• Added automatic linting of treesitter query files (see |ft-query-plugin|).
  Automatic linting can be turned off via >vim

>     autocmd FileType *.scm let b:lint_on = []

Reviewer: ```suggestion
    autocmd FileType query let b:lint_on = []
```

Reviewer: I would prefer Lua snippets in general (for consistency), although Vim is of course more concise here.

Author: still wanted to note that `autocmd FileType query let b:lint_on = []` is not working for me. If you have an alternative suggestion in Lua. I'll paste it here

Reviewer: I'll test. (But this autocommand was wrong in any case, `*.scm` is not a valid FileType pattern.)

> 
> • Enabled treesitter in runtime query files
> <
> • ...

Reviewer: ```suggestion
<
• Treesitter highlighting in query files is enabled by default
```

iterators |luaref-in|.
• Added |vim.keycode()| for translating keycodes in a string.

• Added automatic linting of treesitter query files (see |ft-query-plugin|).
  Automatic linting can be turned off via >lua

> 	vim.g.query_lint_on = {}

Reviewer: ```suggestion
	vim.g.query_lint_on = {}
<
```

(Vim's help parser is laissez-faire, but in our documentation we should strive for "proper" syntax to make the treesitter parser's job easier.)

Author: adopted the change as suggested

> • Enabled treesitter in runtime query files

Reviewer: ```suggestion
• Enabled treesitter highlighting for treesitter query files by default.
```

Author: adopted change as suggested
